intelmq/lib/upgrades.py

```diff
@@ -610,6 +611,37 @@ def v300_defaults_file_removal(defaults, runtime, harmonization, dry_run):
     return messages if messages else changed, defaults, runtime, harmonization
 
 
+def v300_pipeline_file_removal(defaults, runtime, harmonization, dry_run):
+    """
+    Remove the pipeline.conf file
+    """
+    changed = None
+    messages = []
+    pipeline_file = Path(CONFIG_DIR) / "pipeline.conf"
+    if pipeline_file.exists():
+        pipelines = load_configuration(pipeline_file)
+        for bot in runtime:
+            if bot in pipelines:
+                if 'destination-queues' in pipelines[bot]:
+                    destination_queues = pipelines[bot]['destination-queues']
+                    if isinstance(destination_queues, dict):
+                        runtime[bot]['parameters']['destination_queues'] = destination_queues
+                    if isinstance(destination_queues, list):
+                        runtime[bot]['parameters']['destination_queues'] = {'_default': destination_queues}
+                    if isinstance(destination_queues, str):
+                        runtime[bot]['parameters']['destination_queues'] = {'_default': [destination_queues]}
+                if 'source-queue' in pipelines[bot]:
+                    if pipelines[bot]['source-queue'] != f"{bot}-queue":
+                        runtime[bot]['parameters']['source_queue'] = pipelines[bot]['source-queue']
+        if dry_run:
+            print('Would now remove file {pipeline_file!r}.')
+        else:
+            pipeline_file.unlink()
+            changed = True
```

Comment:
```suggestion
        changed = True
```
That's independent of the dry run.